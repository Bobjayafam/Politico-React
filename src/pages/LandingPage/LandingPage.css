.hero {
  background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)), url(./voters.jpg);
  height: 70vh;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
}

.hero-text {
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
}

.hero-text h1 {
  font-size: 48px;
  text-transform: uppercase;
  font-weight: 700;
}

.hero-text p {
  font-size: 20px;
}

.hero button {
  background: #ff9902;
  width: 50%;
  padding: 15px;
  border: 0;
  cursor: pointer;
  border-radius: 5px;
  transition: 0.3s;
  margin-top: 20px;
  font-size: 24px;
  color: #fff;
  font-weight: 600;
}

.hero button:hover {
  opacity: 0.8;
}

.features p {
  text-transform: uppercase;
}

.features-heading {
  font-size: 30px;
}
