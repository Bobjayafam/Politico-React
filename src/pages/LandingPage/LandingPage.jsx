import React from 'react';
import Footer from '../../components/Footer/Footer';
import './LandingPage.css';

const LandingPage = () => (
  <>
    <section className="hero mb-50">
      <div className="hero-text">
        <h1 className="mb-20">Welcome to politico</h1>
        <p className="mb-50">Your Secure online voting platform</p>
        <button type="button">Get started</button>
      </div>
    </section>
    <section className="container">
      <div className="features text-center text-bold mb-20">
        <p className="features-heading">Only three steps needed</p>
      </div>
      <div className="cards text-center">
        <div className="card2">
          <div className="card-label">
            <h1>REGISTER</h1>
            <i className="far fa-registered fa-4x" />
            <p>
              If you are up to the voting age of 18, Register via a simple process. You can register
              on all device type
            </p>
          </div>
        </div>
        <div className="card2">
          <div className="card-label">
            <h1>VOTE</h1>
            <i className="fas fa-person-booth fa-4x" />
            <p>
              Search for the political office and your preferred candidate. Cast your Vote from the
              comfort of your home. No more long queues at polling stations
            </p>
          </div>
        </div>
        <div className="card2">
          <div className="card-label">
            <h1>RESULT</h1>
            <i className="fas fa-poll fa-4x" />
            <p>View Results of a secure and accountable process</p>
          </div>
        </div>
      </div>
    </section>
    <Footer />
  </>
);

export default LandingPage;
