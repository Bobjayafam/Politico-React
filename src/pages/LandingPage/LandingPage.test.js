import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import LandingPage from './LandingPage';

it('Landing Page renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter>
      <LandingPage />
    </MemoryRouter>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
