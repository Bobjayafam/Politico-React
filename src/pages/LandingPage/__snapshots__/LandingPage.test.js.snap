// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "zidmlh",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "zidmlh",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LandingPage>
      <div
        className="main-container"
      >
        <Navbar>
          <nav
            className="navbar container"
          >
            <div
              className="navbar-brand"
            >
              <NavLink
                href="index.html"
                to="/"
              >
                <Route
                  path="\\\\/"
                >
                  <Link
                    aria-current="page"
                    className="active"
                    href="index.html"
                    style={Object {}}
                    to="/"
                  >
                    <a
                      aria-current="page"
                      className="active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <img
                        alt="logo"
                        src="test-file-stub"
                      />
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </div>
            <label
              className="navbar_toggle"
              htmlFor="toggle"
              id="nav-icon"
            >
              ☰
            </label>
            <input
              id="toggle"
              type="checkbox"
            />
            <ul
              className="navbar-menu"
            >
              <li>
                <NavLink
                  href="index.html"
                  to="/"
                >
                  <Route
                    path="\\\\/"
                  >
                    <Link
                      aria-current="page"
                      className="active"
                      href="index.html"
                      style={Object {}}
                      to="/"
                    >
                      <a
                        aria-current="page"
                        className="active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Home
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li>
                <NavLink
                  href="signup.html"
                  to="/signup"
                >
                  <Route
                    path="\\\\/signup"
                  >
                    <Link
                      aria-current={null}
                      href="signup.html"
                      to="/signup"
                    >
                      <a
                        aria-current={null}
                        href="/signup"
                        onClick={[Function]}
                      >
                        SignUp
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li>
                <NavLink
                  href="login.html"
                  to="/login"
                >
                  <Route
                    path="\\\\/login"
                  >
                    <Link
                      aria-current={null}
                      href="login.html"
                      to="/login"
                    >
                      <a
                        aria-current={null}
                        href="/login"
                        onClick={[Function]}
                      >
                        LogIn
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
            </ul>
          </nav>
        </Navbar>
        <section
          className="hero mb-50"
        >
          <div
            className="hero-text"
          >
            <h1
              className="mb-20"
            >
              Welcome to politico
            </h1>
            <p
              className="mb-50"
            >
              Your Secure online voting platform
            </p>
            <button
              type="button"
            >
              Get started
            </button>
          </div>
        </section>
        <section
          className="container"
        >
          <div
            className="features text-center text-bold mb-20"
          >
            <p
              className="features-heading"
            >
              Only three steps needed
            </p>
          </div>
          <div
            className="cards text-center"
          >
            <div
              className="card2"
            >
              <div
                className="card-label"
              >
                <h1>
                  REGISTER
                </h1>
                <i
                  className="far fa-registered fa-4x"
                />
                <p>
                  If you are up to the voting age of 18, Register via a simple process. You can register on all device type
                </p>
              </div>
            </div>
            <div
              className="card2"
            >
              <div
                className="card-label"
              >
                <h1>
                  VOTE
                </h1>
                <i
                  className="fas fa-person-booth fa-4x"
                />
                <p>
                  Search for the political office and your preferred candidate. Cast your Vote from the comfort of your home. No more long queues at polling stations
                </p>
              </div>
            </div>
            <div
              className="card2"
            >
              <div
                className="card-label"
              >
                <h1>
                  RESULT
                </h1>
                <i
                  className="fas fa-poll fa-4x"
                />
                <p>
                  View Results of a secure and accountable process
                </p>
              </div>
            </div>
          </div>
        </section>
        <Footer>
          <footer
            className="text-center mt-20"
          >
            <div
              className="container"
            >
              <span>
                Copyright © 2019 | Politico- An Andela Bootcamp Project
              </span>
            </div>
          </footer>
        </Footer>
      </div>
    </LandingPage>
  </Router>
</MemoryRouter>
`;
